oauth2_http/java/com/google/auth/oauth2/GdchCredentials.java

```diff
@@ -0,0 +1,509 @@
+    this.privateKey = Preconditions.checkNotNull(builder.privateKey);
+    this.serviceIdentityName = Preconditions.checkNotNull(builder.serviceIdentityName);
+    this.tokenServerUri = Preconditions.checkNotNull(builder.tokenServerUri);
+    this.transportFactory =
+        firstNonNull(
+            builder.transportFactory,
+            getFromServiceLoader(HttpTransportFactory.class, OAuth2Utils.HTTP_TRANSPORT_FACTORY));
+    this.transportFactoryClassName = this.transportFactory.getClass().getName();
+    this.caCertPath = builder.caCertPath;
+    this.apiAudience = builder.apiAudience;
+    this.lifetime = builder.lifetime;
+  }
+
+  static class TransportFactoryForGdch implements HttpTransportFactory {
+    HttpTransport transport;
+
+    public TransportFactoryForGdch(String caCertPath) throws IOException {
+      setTransport(caCertPath);
+    }
+
+    @Override
+    public HttpTransport create() {
+      return transport;
+    }
+
+    private void setTransport(String caCertPath) throws IOException {
+      try {
+        InputStream certificateStream = readStream(new File(caCertPath));
+        this.transport =
+            new NetHttpTransport.Builder().trustCertificatesFromStream(certificateStream).build();
+      } catch (IOException e) {
+        throw new IOException(
+            String.format(
+                "Error reading certificate file from CA cert path, value '%s': %s",
+                caCertPath, e.getMessage()),
+            e);
+      } catch (GeneralSecurityException e) {
+        throw new IOException("Error initiating transport with certificate stream.", e);
+      }
+    }
+  }
+
+  /**
+   * Create GDCH service account credentials defined by JSON.
+   *
+   * @param json a map from the JSON representing the credentials.
+   * @param transportFactory HTTP transport factory, creates the transport used to get access
+   *     tokens.
+   * @return the GDCH service account credentials defined by the JSON.
+   * @throws IOException if the credential cannot be created from the JSON.
+   */
+  static GdchCredentials fromJson(Map<String, Object> json, HttpTransportFactory transportFactory)
+      throws IOException {
+    String formatVersion = validateField((String) json.get("format_version"), "format_version");
+    String projectId = validateField((String) json.get("project"), "project");
+    String privateKeyId = validateField((String) json.get("private_key_id"), "private_key_id");
+    String privateKeyPkcs8 = validateField((String) json.get("private_key"), "private_key");
+    String serviceIdentityName = validateField((String) json.get("name"), "name");
+    String tokenServerUriStringFromCreds =
+        validateField((String) json.get("token_uri"), "token_uri");
+
+    /**
+     * The CA cert path for token server side TLS, certificate verification. If the token server
+     * uses well known CA, then this parameter can be `None`.
+     */
+    String caCertPath = (String) json.get("ca_cert_path");
```

Comment:
this could be a "package private" static (not public) logic and acessed from Google credential to create a correct transport

Reply:
you can have a method like "getGdchTransport(FileContents)"

Reply:
scratch the above, let's have another overload without the parameter that creates the transport and calls this method  with it

Reply:
done